### ISSUE TITLE
`lg_exp_stringify()` (and also `lg_exp_*()`)

### ISSUE THREAD

**Issue by ampli** [opencog/link-grammar#1222]
https://github.com/opencog/link-grammar/blob/bdb90344f8474157037c9bf3f6b5df570aecaefd/link-grammar/dict-common/dict-structures.h#L98-L107

The `lg_exp_stringify()` returns a pointer to a static TLS memory. I did it this way because it is used internally in many places and it was inconvenient to use a temporary variable + `free()`. However, this is not consistent with the rest of the API functions, which don't change the memory pointed by existing pointers on subsequent calls.

So I propose to change it to return newly allocated memory that needs to be freed by the API user. (For internal use, I would define a new function `exp_stringify()` returning a static TLS memory, and the API user can do the same if desired.)

There is also a question of how to free the result. The rest of the API uses special memory-freeing functions that usually just call `free()`. I guess this is for future flexibility in case a more complex freeing would be needed. But `free()` seems to always be fine because future added allocations could be done in one contiguous block. In case a special memory-freeing function is desired, I couldn't think of a good name for it... (`lg_exp_free_stringify()` is not good...).

A similar API-design problem exists with the new `lg_exp_resolve()` (which returns an expression in one memory block),
and the existing experimental functions `sentence_unused_disjuncts()` and `disjunct_expression()` (`free()` can be used for all of them).

While writing this I noted that the `lg_exp_operand*()` functions don't use `const` on their parameter and returned result. I cannot think of a reason for that, and maybe this is unintentional. I propose to fix that.

To sum up:
1. Proposal: Change `lg_exp_stringify()` to return new memory.
2. Question: Should we define special memory-freeing functions for all the API functions (listed above) that return new memory that has to be freed?
3. Proposal: Use `const` for `lg_exp_operand*()` parameter and result.

**Comment by linas** [comment]
I merged #1224 without thinking very much about it.  Perhaps its just late at night, and I don't want to think about it :-/ 

const for the static inline is fine; since they are inlined, any imagined future changes won't be binary compatible.  I won't sweat bullets because this is not glibc.

grepping through my code base, I see no external users of stringify -- I thought there was one, but I don't see it.
